File: Maximum Meetings in One Room[22-December-2023]/Solution.java
//        >> JAVA CODE <<



class Solution {
    static class Pair{
        int st,end,index;
        Pair(int st,int end,int index)
        {
            this.st=st;
            this.end=end;
            this.index=index;
        }
    }
    public static ArrayList<Integer> maxMeetings(int N, int[] S, int[] F) {
        
        ArrayList<Pair> meetings=new ArrayList<>();
        
        for(int i=0;i<N;i++)
        {
            meetings.add(new Pair(S[i],F[i],i+1));
        }
        
        Collections.sort(meetings,(a,b)->Integer.compare(a.end,b.end));
        
        ArrayList<Integer> ans=new ArrayList<>();
        Pair prev=null;
        for(int i=0;i<N;i++)
        {
            if(i==0)
            {
                ans.add(meetings.get(i).index);
                prev=meetings.get(i);
            }
            else
            {
                if(meetings.get(i).st>prev.end)
                {
                    ans.add(meetings.get(i).index);
                    prev=meetings.get(i);
                }
            }
        }
        
        Collections.sort(ans);
        
            return ans;
        
    }
}
        

File: Maximum Meetings in One Room[22-December-2023]/Solution.cpp
//         >> CPP CODE <<

class Pair {
public:
    int st, end, index;

    Pair(int st, int end, int index) : st(st), end(end), index(index) {}
};

class Solution {
public:
    vector<int> maxMeetings(int N, vector<int>& S, vector<int>& F) {
        vector<Pair> meetings;

        for (int i = 0; i < N; i++) {
            meetings.push_back(Pair(S[i], F[i], i + 1));
        }

        sort(meetings.begin(), meetings.end(), [](const Pair& a, const Pair& b) {
            return a.end < b.end;
        });

        vector<int> ans;
        Pair* prev = nullptr;

        for (int i = 0; i < N; i++) {
            if (i == 0) {
                ans.push_back(meetings[i].index);
                prev = &meetings[i];
            } else {
                if (meetings[i].st > prev->end) {
                    ans.push_back(meetings[i].index);
                    prev = &meetings[i];
                }
            }
        }

        sort(ans.begin(), ans.end());

        return ans;
    }
};